### PR TITLE
Download all button on submission

### LINKED ISSUE #1571: Download All on Submission Page
Can we add a "Download All" button on the (student) Submission page for electronic gradeables?

### PR DESCRIPTION
**Pull request by ropern123**
Adds a button on the student homework page that students can use to download all their files in a zip

closes #1571

### CODE REVIEW

File: site/app/controllers/MiscController.php [hunk 1]
```diff
@@ -311,6 +322,10 @@ private function downloadZip() {
         header("Expires: 0");
         readfile("$zip_name");
         unlink($zip_name); //deletes the random zip file
+
+        if(!$this->core->getUser()->accessGrading()) {
+            // make sure that the user id matches the requested id, and check the version number
+        }
```
Comment (KevinMackenzie) [inline site/app/controllers/MiscController.php]: What are these 4 lines for?

File: site/public/js/server.js [hunk 2]
```diff
@@ -997,7 +997,35 @@ function check_server(url) {
...
+            'component': 'misc',
+            'page': 'download_zip',
+            'dir': 'submissions',
+            'gradeable_id': grade_id,
+            'user_id': user_id
+        });
+    }
+    else {
+        window.location = buildUrl({
+            'component': 'misc',
```
Comment (KevinMackenzie) [inline site/public/js/server.js]: factor the assignment of `window.location` and the call to `buildUrl` out of the if statement

File: site/app/controllers/MiscController.php [hunk 3]
```diff
@@ -37,12 +37,19 @@ public function run() {
...
         // only allow zip if it's a grader
         if ($is_zip) {
-            $error_string="only graders may access zip files";
-            return ($this->core->getUser()->accessGrading());
+            $error_string="only graders may access zip files of other students";
+            $can_download = (
+                $gradeable !== null
+                && !$gradeable->useVcsCheckout()
+                && $gradeable->getStudentDownload()
+                && !($gradeable->getCurrentVersionNumber() !== $gradeable->getActiveVersion() && !$gradeable->getStudentAnyVersion())
```
Comment (KevinMackenzie) [inline site/app/controllers/MiscController.php]: Would be more clear what this is doing if it was 
` && ($gradeable->getCurrentVersionNumber() === $gradeable->getActiveVersion() || $gradeable->getStudentAnyVersion())` instead

Comment (ropern123) [reply site/app/controllers/MiscController.php]: This logic was taken from the code that decides whether or not a version is downloadable. I'll go ahead and change that and this.

File: site/app/controllers/MiscController.php (hunk 1 above)
Comment (ropern123) [reply site/app/controllers/MiscController.php]: whoops

**Pull request merged**